openbb_platform/providers/fmp/openbb_fmp/models/government_trades.py

```diff
@@ -0,0 +1,163 @@
+            "all": ["senate-disclosure", "senate-trading"],
+        }
+        api_key = credentials.get("fmp_api_key") if credentials else ""
+
+        async def get_one(url):
+            # 指定要移除的键
+            keys_to_remove = [
+                "comment",
+                "district",
+                "capitalGainsOver200USD",
+                "disclosureYear",
+            ]
+            # 指定要重命名的键，格式为 {原键: 新键}
+            keys_to_rename = {"dateRecieved": "date", "disclosureDate": "date"}
+            """Get data for the given symbol."""
+
+            result = await amake_request(url, response_callback=response_callback, **kwargs)
+            # 处理数据
+            processed_list = []
+            for entry in result:
+                # 创建新的字典用于存储处理后的数据
+                new_entry = {k: v for k, v in entry.items() if k not in keys_to_remove}
+                # 重命名指定的键
+                for old_key, new_key in keys_to_rename.items():
+                    if old_key in new_entry:
+                        new_entry[new_key] = new_entry.pop(old_key)
+                processed_list.append(new_entry)
+            if not processed_list or len(processed_list) == 0:
+                warn(f"Symbol Error: No data found for {url}")
+            if processed_list:
+                results.extend(processed_list)
+
+        if symbols:
+            urls_list = []
+            for symbol in symbols:
+                query.symbol = symbol
+                url = [
+                    create_url(
+                        4,
+                        f"{i}",
+                        api_key=api_key,
+                        query=query,
+                        exclude=["chamber", "limit"],
+                    )
+                    for i in chamber_url_dict[query.chamber]
+                ]
+                urls_list.extend(url)
+            await asyncio.gather(*[get_one(url) for url in urls_list])
+        else:
+            urls_list = []
+            pages = math.ceil(query.limit / 100)
+            for page in range(pages):
+                query.page = page
+                url = [
+                    create_url(
+                        4,
+                        f"{i}-rss-feed",
+                        api_key=api_key,
+                        query=query,
+                        exclude=["chamber", "limit"],
+                    )
+                    for i in chamber_url_dict[query.chamber]
+                ]
+                urls_list.extend(url)
+            await asyncio.gather(*[get_one(url) for url in urls_list])
+        if not results:
+            raise EmptyDataError("No data returned for the given symbol.")
+
+        return results
+
+    @staticmethod
+    def transform_data(
+        query: FMPGovernmentTradesQueryParams, data: List[Dict], **kwargs: Any
+    ) -> List[FMPGovernmentTradesData]:
+        """Return the transformed data."""
+        return [FMPGovernmentTradesData(**d) for d in data]
```

Comment:
At this stage, we should sort the data by date because the `gather` process will be appending as they come. 

There's also some empty strings being passed in a few fields. `None` is preferred for `Optional[str]` because evaluating for None is more consistent than checking truthiness. For example, it's not uncommon to receive `"''"`, which would pass truthiness.

Example of empty strings, "owner="


```
FMPGovernmentTradesData(symbol=AMZN, date=2024-11-22, transaction_date=2024-11-20, representative=Marjorie Taylor Greene, link=https://disclosures-clerk.house.gov/public_disc/ptr-pdfs/2024/20026257.pdf, owner=, asset_type=None, asset_description=Amazon.com Inc, type=Purchase, amount=$1,001 - $15,000, comment=None)
```